### PR TITLE
Add the Managarm Handbook into docs

### PR DESCRIPTION
**Pull request by unpause**
This PR merges the [mdBook](https://github.com/rust-lang/mdBook) handbook for Managarm into the repository which can be found in the `docs` folder.

This addresses the recent discussions on the Discord about the lacking documentation of the project.

The documentation can be built locally by executing `mdbook build` in the `docs` directory, the output can then be found in the `docs/book` directory.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Managarm Documentation
+
+![github pages](https://github.com/unpause/managarm-docs/workflows/github%20pages/badge.svg)
```
Comment (avdgrinten) [inline docs/README.md:3]: This should probably be removed.

File: docs/src/index.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Managarm Handbook
+
+<center>
+<img src="https://avatars2.githubusercontent.com/u/38985399?s=200&v=4">
+</center>
```
Comment (avdgrinten) [inline docs/src/index.md:5]: This links directly to the current avatar of the org?

**Comment by unpause** [comment]
I removed the GitHub pages badge and added the logo image file to the repository.

File: docs/src/index.md (hunk 2 above)
Comment (unpause) [reply docs/src/index.md]: i added the image to the repo itself in the newest commit.

File: docs/README.md (hunk 1 above)
Comment (unpause) [reply docs/README.md]: removed in latest commit.

**Comment by avdgrinten** [comment]
Thank you very much for the PR!

**Pull request merged**